industrial_ci/src/tests/source_tests.sh

```diff
@@ -33,14 +33,31 @@ function run_clang_tidy {
     local db=$4
     shift 4
 
+    local build; build="$(dirname "$db")"
+    local name; name="$(basename "$build")"
+    ici_time_start "clang_tidy_check_$name"
+
+    # create an array of all files listed in $db filtered by the source tree
     mapfile -t files < <(grep -oP "(?<=\"file\": \")($regex)(?=\")" "$db")
+    local num_all_files; num_all_files="${#files[@]}"
+    if [ -n "$CLANG_TIDY_BASE_REF" ] ; then
+        echo "Filtering for files that actually changed since $CLANG_TIDY_BASE_REF"
+        # Need to run git in actual source dir:  $files[@] refer to source dir and $PWD is read-only
+        local src_dir
+        src_dir=$(grep -oP "(?<=CMAKE_HOME_DIRECTORY:INTERNAL=).*" "$build/CMakeCache.txt")
+        pushd "$src_dir" > /dev/null || true
+        if git fetch -q origin "$CLANG_TIDY_BASE_REF" ; then  # git might fail, e.g. operating on catkin_tools_prebuild
```

Comment:
stderr should be written `/dev/null` to get rid of https://github.com/rhaschke/rviz/runs/2313129434?check_suite_focus=true#step:7:209:

```
 fatal: not a git repository (or any parent up to mount point /home/runner)
 Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

Reply:
Done. Corresponding example build is retriggered: https://github.com/rhaschke/rviz/runs/2327098690